isis/CMakeLists.txt

```diff
@@ -490,6 +509,90 @@ add_custom_target(incs ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+add_dependencies(isis3 cnetref)
+
+add_custom_target(cnetstats ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/cnetstats/* ${CMAKE_BINARY_DIR}/appdata/templates/cnetstats/)
+add_dependencies(isis3 cnetstats)
+
+add_custom_target(cnetvalid ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/cnet_validmeasure/* ${CMAKE_BINARY_DIR}/appdata/templates/cnet_validmeasure/)
+add_dependencies(isis3 cnetvalid)
+
+add_custom_target(controlnetworks ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/controlnetworks/* ${CMAKE_BINARY_DIR}/appdata/templates/controlnetworks/)
+add_dependencies(isis3 controlnetworks)
+
+add_custom_target(findfeatures ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/findfeatures/* ${CMAKE_BINARY_DIR}/appdata/templates/findfeatures/)
+add_dependencies(isis3 findfeatures)
+
+add_custom_target(fullcnetdiff ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/fullcnetdiff/* ${CMAKE_BINARY_DIR}/appdata/templates/fullcnetdiff/)
+add_dependencies(isis3 fullcnetdiff)
+
+add_custom_target(jigsaw ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/jigsaw/* ${CMAKE_BINARY_DIR}/appdata/templates/jigsaw/)
+add_dependencies(isis3 jigsaw)
+
+add_custom_target(kernels ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/kernels/* ${CMAKE_BINARY_DIR}/appdata/templates/kernels/)
+add_dependencies(isis3 kernels)
+
+add_custom_target(photometry ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/photometry/* ${CMAKE_BINARY_DIR}/appdata/templates/photometry/)
+add_dependencies(isis3 photometry)
+
+add_custom_target(icons ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/images/icons/* ${CMAKE_SOURCE_DIR}/appdata/images/icons/licenses/* 
+  ${CMAKE_BINARY_DIR}/appdata/images/icons)
+add_dependencies(isis3 icons)
+
+add_custom_target(targets ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/images/targets/* ${CMAKE_BINARY_DIR}/appdata/images/targets)
+add_dependencies(isis3 targets)
+
+add_custom_target(maps ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/maps/* ${CMAKE_BINARY_DIR}/appdata/templates/maps)
+add_dependencies(isis3 maps)
+
+add_custom_target(labels ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/labels/* ${CMAKE_BINARY_DIR}/appdata/templates/labels)
+add_dependencies(isis3 labels)
+
+add_custom_target(hidtmgen ALL COMMAND ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_SOURCE_DIR}/appdata/templates/hidtmgen/* ${CMAKE_BINARY_DIR}/appdata/templates/hidtmgen/)
+add_dependencies(isis3 hidtmgen)
+
```

Comment:
Are the custom targets here just trying to be smarter about copying things into the build directory?

Reply:
The ISIS3 to ISIS came along for the ride. It was used to make sure we found and changed every reference.

Custom targets, yes, but I'll let Kristin weigh in for more details

Reply:
@acpaquette The goal of the several custom targets was the following:

(1) Get everything copied over into the correct sub-directory of the build (equivalent of cp -r isis/appdata/templates $ISISROOT/appdata, for example) Retaining the directory structure was necessary.

(2) During subsequent builds, re-copy the file from the source to the build if it has changed since the first build.

Things I tried: 
I didn't just use cmake's "execute command" with `cp -r` or `file(COPY...` because then (2) doesn't work. `configure_file` works great for (2) but only works for individual files, not lists of files. 

I'm sure there is a better way to do this and am very open to other ideas / suggestions. I am still very much a cmake novice.